src/logic/map_objects/tribes/ship.cc

```diff
@@ -883,23 +912,35 @@ WalkingDir Ship::get_scouting_direction() const {
 /// @note only called via player command
 void Ship::exp_construct_port(Game& game, const Coords& c) {
 	assert(expedition_);
-	const MapObject* mo = game.map().get_fcoords(c).field->get_immovable();
-	if ((mo != nullptr) && mo->descr().type() >= MapObjectType::BUILDING) {
-		// Another expedition ship (or an enemy player) was a second faster
-		set_ship_state_and_notify(
-		   ShipStates::kExpeditionWaiting, NoteShip::Action::kDestinationChanged);
-		return;
+	// recheck ownership before setting the csite
+	Map* map = game.mutable_map();
+	if (!expedition_->seen_port_buildspaces.empty()) {
+		const FCoords fc = map->get_fcoords(expedition_->seen_port_buildspaces.front());
+		if (!can_build_port_here(get_owner()->player_number(), *map, fc)) {
+			set_ship_state_and_notify(
+			   ShipStates::kExpeditionWaiting, NoteShip::Action::kDestinationChanged);
+			send_message(
+			   game, _("Port Space"), _("Port Space Lost"),
+			   _("A discovered port build space is not available for building a port anymore."),
+			   "images/wui/editor/fsel_editor_set_port_space.png");
+			expedition_->seen_port_buildspaces.clear();
+			return;
+		}
+	} else {
+		log_warn_time(
+		   game.get_gametime(), "Expedition list of seem üport spaces is unexpectedly empty!\n");
```

Comment:
```suggestion
		   game.get_gametime(), "Expedition list of seen port spaces is unexpectedly empty!\n");
```

(also the other one – see? its worth the function ;)